sdk-core-protos/src/lib.rs

```diff
@@ -209,6 +210,24 @@ pub mod coresdk {
         }
 
         impl ActivityResolution {
+            /// A helper function for cases when you want to extract an activity's payload if it
+            /// completed successfully or return an error for all other outcomes. It's essentially
+            /// an alternative to [`unwrap_ok_payload`][`Self::unwrap_ok_payload`] that uses neither
+            /// `unwrap()` nor `panic!`.
+            pub fn success_payload_or_error(self) -> Result<Option<Payload>, anyhow::Error> {
+                let Some(status) = self.status else {
+                    return Err(anyhow!("activity completed without a status"));
+                };
+
+                match status {
+                    activity_resolution::Status::Completed(success) => Ok(success.result),
+                    status => Err(anyhow!(
+                        "activity resulted in a non-completed status: {:?}",
```

Comment:
I'd re-use the same string from below here and then make `unwrap_ok_payload` call this and just unwrap it